setuptools/msvc.py

```diff
@@ -212,6 +221,19 @@ def msvc14_get_vc_env(plat_spec):
         raise
 
 
+def msvc14_gen_lib_options(*args, **kwargs):
+    """
+    Patched "distutils._msvccompiler.gen_lib_options" for fix 
+    compatibility between "numpy.distutils" and "distutils._msvccompiler"
+    (for Numpy < 1.11.2)
+    """
+    if "numpy.distutils" in sys.modules:
+        import numpy as np
+        if StrictVersion(np.__version__) < StrictVersion('1.11.2'):
```

Comment:
You cannot use ``StrictVersion`` here, that's a broken braindead thing. You will get something like

    ValueError: invalid version number '1.11.1rc1+fa248ad'

for any development version. This is true for many projects, not just numpy. From numpy 1.9.0 on ``numpy.lib.NumpyVersion`` is available, which does proper version comparisons. IIRC something similar that does the right thing for any PEP 440 compliant version string is present in ``distlib`` or another PyPA project.